src/components/common/YourCompany/OfficeAddresses.vue

```diff
@@ -779,7 +793,26 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
   @Watch('isEditing', { immediate: true })
   private onEditingChanged (val: boolean): void {
     this.setEditingOfficeAddresses(val) // Used for Correction Flags
-    this.setValidComponent({ key: 'isValidAddress', value: !this.isEditing })
+    this.updateValidity()
+  }
+
+  /**
+   * Updates component validity when any address validity has changed.
+   * Also called when addresses are set/reset.
+   */
+  @Watch('mailingAddressValid')
+  @Watch('deliveryAddressValid')
+  @Watch('recMailingAddressValid')
+  @Watch('recDeliveryAddressValid')
+  private updateValidity (): void {
+    const isValid = (
+      !this.isEditing &&
+      this.mailingAddressValid &&
+      this.deliveryAddressValid &&
+      this.recMailingAddressValid &&
+      this.recDeliveryAddressValid
+    )
+    this.setValidComponent({ key: 'isValidAddress', value: isValid })
```

Comment:
Now, this component's validity flag looks at not only whether it's open for editing, but also whether all addresses are valid. (Invalid error styling won't display until the global validate flag is enabled.)

This allows this component to block the user from going to the Review and Confirm page if invalid.